src/Workspaces/Core/Portable/Execution/ChecksumTreeCollection.TreeNodes.cs

```diff
@@ -179,6 +247,59 @@ public IChecksumTreeNode GetOrCreateSubTreeNode<TKey>(TKey key)
                 return GetOrCreateChecksumObjectAsync(key, value, kind, valueGetterAsync, cancellationToken);
             }
 
+            protected void AppendChecksumObjects(
+                ImmutableDictionary<Checksum, ChecksumObject>.Builder builder,
+                HashSet<Checksum> checksums,
+                int itemCount,
+                IEnumerable<Checksum> itemChecksums,
+                Func<Checksum, ChecksumObject> itemGetter, CancellationToken cancellationToken)
+            {
```

Comment:
what's the concept here.  why are there two sets of checksums, and you only pick the smaller set?